desc/integrals/singularities.py

```diff
@@ -805,12 +806,15 @@ def _kernel_1_over_r(eval_data, source_data, diag=False):
 def _kernel_nr_over_r3(eval_data, source_data, diag=False):
     """Returns n ⋅ −∇G(x,x') = n ⋅ (x−x')|x−x'|⁻³."""
     dx = _dx(eval_data, source_data, diag)
-    n = rpz2xyz_vec(source_data["n_rho"], phi=source_data["phi"])
+    # Need to use e^rho*sqrt(g) to pass Green's ID test.
+    # Fourier spectrum is much more concentrated than n_rho for some reason.
```

Comment:
otherwise the w7x one fails, which is interesting because there's many nodes and no truncation.